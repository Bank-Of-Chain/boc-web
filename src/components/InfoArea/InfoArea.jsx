import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// nodejs library that concatenates classes
import classNames from "classnames";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";

import styles from "./infoStyle";

const useStyles = makeStyles(styles);

export default function InfoArea(props) {
  const classes = useStyles();
  const { title, description, iconColor, vertical, style } = props;
  const iconWrapper = classNames({
    [classes.iconWrapper]: true,
    [classes[iconColor]]: true,
    [classes.iconWrapperVertical]: vertical,
  });
  return (
    <div className={classes.infoArea} style={style}>
      <div className={iconWrapper}>{props.icon}</div>
      <div className={classes.descriptionWrapper}>
        <h4 className={classes.title}>{title}</h4>
        <p className={classes.description}>{description}</p>
      </div>
    </div>
  );
}

InfoArea.defaultProps = {
  iconColor: "gray",
};

InfoArea.propTypes = {
  icon: PropTypes.element.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.array.isRequired || PropTypes.string.isRequired,
  iconColor: PropTypes.oneOf([
    "primary",
    "warning",
    "danger",
    "success",
    "info",
    "rose",
    "gray",
  ]),
  vertical: PropTypes.bool,
};
