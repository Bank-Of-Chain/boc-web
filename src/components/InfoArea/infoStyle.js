import {
  primaryColor,
  warningColor,
  dangerColor,
  successColor,
  infoColor,
  roseColor,
  grayColor,
  title,
} from "../../assets/jss/material-kit-react.js"

const infoStyle = {
  iconWrapper: {
    float: "left",
    marginTop: "24px",
    marginRight: "10px",
  },
  primary: {
    color: primaryColor,
  },
  warning: {
    color: warningColor,
  },
  danger: {
    color: dangerColor,
  },
  success: {
    color: successColor,
  },
  info: {
    color: infoColor,
  },
  rose: {
    color: roseColor,
  },
  gray: {
    color: grayColor,
  },
  icon: {
    width: "36px",
    height: "36px",
  },
  descriptionWrapper: {
    color: grayColor,
    overflow: "hidden",
  },
  title: {
    ...title,
    color: "#fff",
    fontSize: "1.5rem",
  },
  description: {
    color: "#fff",
    overflow: "hidden",
    marginTop: "0px",
    fontSize: "14px",
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    lineHeight: "1.5em",
    textAlign: "left",
  },
  iconWrapperVertical: {
    float: "none",
  },
  iconVertical: {
    width: "61px",
    height: "61px",
  },
}

export default infoStyle
