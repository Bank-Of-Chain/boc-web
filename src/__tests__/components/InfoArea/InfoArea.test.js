import React from "react";
import { render } from "@testing-library/react";
import InfoArea from "../../../components/InfoArea/InfoArea";

test("InfoArea Component Render", () => {
  const { asFragment } = render(<InfoArea icon={<span>test</span>} />);
  expect(asFragment()).toMatchSnapshot();
});
