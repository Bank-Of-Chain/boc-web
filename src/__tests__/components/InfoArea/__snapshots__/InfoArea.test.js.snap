// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoArea Component Render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-infoArea-15"
  >
    <div
      class="makeStyles-iconWrapper-1 makeStyles-gray-8"
    >
      <span>
        test
      </span>
    </div>
    <div
      class="makeStyles-descriptionWrapper-10"
    >
      <div
        class="makeStyles-title-11"
      >
        test
      </div>
      <p
        class="makeStyles-description-12"
      >
        description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
